robottelo/hosts.py

```diff
@@ -2054,9 +2054,16 @@ def get_rhsso_groups_details(self, group_name):
         return query_group[0]
 
     def upload_rhsso_entity(self, json_content, entity_name):
-        """Helper method upload the entity json request as file on RHSSO Server"""
+        """Helper method to upload the RHSSO entity file on RHSSO Server.
+        Overwrites already existing file with the same name.
+        """
         with open(entity_name, "w") as file:
             json.dump(json_content, file)
+        # Before uploading a file, remove the file of the same name. In sftp_write,
+        # if uploading a file of length n when there was already uploaded a file with
+        # the same name of length m, for n<m, only first n characters are replaced by
+        # the characters in new file and the rest is left as it is.
```

Comment:
@JacobCallahan Could you comment on whether this is the expected behavior of `sftp_write`? We are talking about `Session.sftp_write` not the `ContainerSession` one.